### DISCUSSION TITLE
Asking for Suggestions related to Anomalib usage

### DISCUSSION THREAD

**Discussion by Sahil1776** [open-edge-platform/anomalib#1416] [Q&A]
I have been trying to use Anomalib for my project.

I have a dataset of images with 4 classes with increasing order of abnormality. Here are the classes:
- Score_1 (Normal)
- Score_2 (Little bad)
- Score_3 (Bad)
- Score_4 (Worse)

I want to classify these images. I have an unlabelled Test set which I want to use as well.
I want to implement the EfficientAD algorithm.

I want to ask…how to use my dataset for this? I have a training set with 4 classes and an unlabelled test set. I want to use the test set as well.
At the end, I want some kind of metric (like reconstruction error) that I can use classify the 4 different levels of abnormality. My goal is Classification. The images are of the same thing which different level of abnormality. 

I highly appreciate if anyone can provide some suggestions.
Thank you

**Comment by ashwinvaidya17** [comment]
You can have a look at our [Folder Dataset](https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/data/folder.py). This [notebook](https://github.com/openvinotoolkit/anomalib/blob/main/notebooks/100_datamodules/103_folder.ipynb) might help you.

Since the currently available models use only the normal images to learn the distribution for outlier detection, we do not distinguish between the anomaly types.

**Comment by abc-125** [comment]
You can try a regular multiclass classification model instead of anomaly detection for this task. Especially if you have a reasonable amount of bad images and they look similar enough inside one class.

**Reply by Sahil1776** [reply]
I have tried it already. The images are of the same thing with the same abnormality in different stages. The classifier ends up predicting the other class. The accuracy was really bad. 

This is the case where images are of the same thing with the same abnormality. The abnormality comes in different stages. I need to classify that stage.
